store/storetest/channel_store.go

```diff
@@ -563,36 +563,28 @@ func testChannelStoreGetByName(t *testing.T, ss store.Store) {
 	o1.Type = model.CHANNEL_OPEN
 	store.Must(ss.Channel().Save(&o1, -1))
 
-	r1 := <-ss.Channel().GetByName(o1.TeamId, o1.Name, true)
-	if r1.Err != nil {
-		t.Fatal(r1.Err)
-	} else {
-		if r1.Data.(*model.Channel).ToJson() != o1.ToJson() {
-			t.Fatal("invalid returned channel")
-		}
-	}
-
-	if err := (<-ss.Channel().GetByName(o1.TeamId, "", true)).Err; err == nil {
-		t.Fatal("Missing id should have failed")
-	}
+	result := <-ss.Channel().GetByName(o1.TeamId, o1.Name, true)
+	require.NotNil(t, result)
+	assert.Nil(t, result.Err)
+	assert.Equal(t, o1.ToJson(), result.Data.(*model.Channel).ToJson(), "invalid returned channel")
 
-	r1 = <-ss.Channel().GetByName(o1.TeamId, o1.Name, false)
-	if r1.Err != nil {
-		t.Fatal(r1.Err)
-	}
-	if r1.Data.(*model.Channel).ToJson() != o1.ToJson() {
-		t.Fatal("invalid returned channel")
-	}
+	result = (<-ss.Channel().GetByName(o1.TeamId, "", true))
+	require.NotNil(t, result)
+	assert.NotNil(t, result.Err, "Missing id should have failed")
```

Comment:
This should be a require to be equivalent to the previous version.

Reply:
My guideline is that I use `assert` when ever I can and use `require` when a test is not able to run continue. What do you think about this guideline?

Reply:
I think is correct, but wasn't replicating the previous behavior. I guess in this cases makes sense to change the previous behavior to allow to continue testing the rest of the cases.